src/averageAges.js

```diff
@@ -54,8 +75,28 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const familyDiff = [];
+  const mothers = people.filter(woman => {
+    return woman.sex === 'f'
+    && people.some(person => person.mother === woman.name);
+  });
+  const children = people.filter(person => {
+    return mothers.some(mother => person.mother === mother.name)
+    && (onlyWithSon === undefined
+      ? true
+      : person.sex === 'm');
+  });
+
+  children.forEach(child => {
+    const foundMother = mothers.find(mom => mom.name === child.mother);
```

Comment:
why do you need mothers if you are anyway finding in it, code will be simpler if you remove it and just use `people`

Reply:
That’s

Reply:
I compere length og people and length of mothers(39 vs 14). So, when I use forEach to array "mothers" i do less itterations

Reply:
Btw, when you were looking for moms, you already made one extra circe in second degree for it)